modules/videoio/src/cap.cpp

```diff
@@ -117,6 +117,26 @@ bool VideoCapture::open(const String& filename, int apiPreference, const std::ve
 
     const VideoCaptureParameters parameters(params);
     const std::vector<VideoBackendInfo> backends = cv::videoio_registry::getAvailableBackends_CaptureByFilename();
+
+    if(apiPreference != CAP_ANY)
+    {
+        //Check if the backend generally is available but can't be used to capture by filename
+        bool foundFilenameBackend = false;
+        for (size_t i = 0; i < backends.size(); i++)
+        {
+            if((foundFilenameBackend = (apiPreference == backends[i].id)))
+                break;
+        }
+
+        if(!foundFilenameBackend && cv::videoio_registry::hasBackend(static_cast<VideoCaptureAPIs>(apiPreference)))
+        {
+            CV_CAPTURE_LOG_DEBUG(NULL,
```

Comment:
I believe `CV_LOG_WARNING` could be used in this case directly (without preconditions).

Reply:
Alright